pix-editor/app/components/form/whitelisted-url.js

```diff
@@ -0,0 +1,237 @@
+      this.checkType.setValue(this.args.initialCheckType);
+    } else {
+      this.checkType.setValue(this.checkTypeOptions[0].value);
+    }
+    this.checkType.validate();
+    this.checkFormValidity();
+  }
+
+  @action
+  async onSubmitClicked(event) {
+    event.preventDefault();
+    this.errorMessages.length = 0;
+    this.isSubmitting = true;
+    const formData = {
+      url: this.url.getValueForSubmit(),
+      comment: this.comment.getValueForSubmit(),
+      relatedSkillNames: this.relatedSkillNames.getValueForSubmit(),
+      checkType: this.checkType.getValueForSubmit(),
+    };
+    try {
+      await this.args.onFormSubmitted(formData);
+    } catch (err) {
+      this.errorMessages.pushObjects(err.message.split('\n'));
+    } finally {
+      this.isSubmitting = false;
+    }
+  }
+
+  @action
+  updateUrl(event) {
+    this.url.setValue(event.target.value);
+    this.validateUrl();
+  }
+
+  @action
+  validateUrl() {
+    this.url.validate();
+    this.checkFormValidity();
+  }
+
+  @action
+  updateComment(event) {
+    this.comment.setValue(event.target.value);
+    this.comment.validate();
+    this.checkFormValidity();
+  }
+
+  @action
+  updateRelatedSkillNames(event) {
+    this.relatedSkillNames.setValue(event.target.value);
+    this.validateRelatedSkillNames();
+  }
+
+  @action
+  validateRelatedSkillNames() {
+    this.relatedSkillNames.validate();
+    this.checkFormValidity();
+  }
+
+  @action
+  updateCheckType(value) {
+    this.checkType.setValue(value);
+    this.checkType.validate();
+    this.checkFormValidity();
+  }
+
+  checkFormValidity() {
+    this.isFormInvalid = !this.url.isValid || !this.relatedSkillNames.isValid || !this.checkType.isValid || !this.comment.isValid;
+  }
+}
+
+class FormField {
+  static STATES = {
+    DEFAULT: 'default',
+    SUCCESS: 'success',
+    ERROR: 'error',
+  };
+
+  @tracked state;
+  @tracked value;
+  @tracked errorMessage;
+
+  constructor({ state = FormField.STATES.DEFAULT, value = '', errorMessage = '' } = {}) {
+    this.state = state;
+    this.value = value;
+    this.errorMessage = errorMessage;
+  }
+
+  get isError() {
+    return this.state === FormField.STATES.ERROR;
+  }
+
+  get isValid() {
+    return this.state === FormField.STATES.SUCCESS;
+  }
+
+  setValue(value) {
+    this.value = value;
+  }
+
+  validate() { throw new Error('implement me');}
+
+  getValueForSubmit() { throw new Error('implement me');}
+}
+
+class UrlField extends FormField {
+  constructor() {
+    super();
+  }
+
+  validate() {
+    if (this.value.trim().length > 0) {
+      this.state = FormField.STATES.SUCCESS;
+      this.errorMessage = '';
+    } else {
+      this.state = FormField.STATES.ERROR;
+      this.errorMessage = 'L\'URL est obligatoire';
+    }
+  }
+
+  getValueForSubmit() { return this.value.trim(); }
+}
+
+class CommentField extends FormField {
+
+  get isValid() { return true; };
+
+  validate() { this.state = FormField.STATES.SUCCESS; }
```

Comment:
J'vais faire ma reloue peut-être, mais ici, on veut pas s'assurer d'avoir des string ? De sanitizer ça aussi ? Histoire de faire dans le hbs `{{on "focusout" this.validateComment}}` ?

Reply:
Le commentaire est optionnel, et vu que c'est un champ textarea, la valeur sera toujours de type string, donc pas besoin normalement 👀